x-pack/plugins/secops/public/components/load_more_table/index.tsx

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiBasicTable,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPopover,
+  EuiTitle,
+} from '@elastic/eui';
+import { isEmpty } from 'lodash/fp';
+import React from 'react';
+import styled from 'styled-components';
+
+import { EventItem } from '../../../common/graphql/types';
+import { LoadingPanel } from '../loading';
+
+export interface HoryzontalBarChartData {
+  x: number;
+  y: string;
+}
+
+export interface ItemsPerRow {
+  text: string;
+  numberOfRow: number;
+}
+
+interface BasicTableProps {
+  // tslint:disable-next-line:no-any
+  pageOfItems: any[];
+  columns: Columns[];
+  title: string | React.ReactNode;
+  loading: boolean;
+  loadingTitle?: string;
+  hasNextPage: boolean;
+  loadMore: () => void;
+  updateLimitPagination: (limit: number) => void;
+  itemsPerRow?: ItemsPerRow[];
+  limit: number;
+}
+
+interface BasicTableState {
+  isPopoverOpen: boolean;
+}
+
+export interface Columns {
+  field?: string;
+  name: string;
+  isMobileHeader?: boolean;
+  sortable?: boolean;
+  truncateText?: boolean;
+  hideForMobile?: boolean;
+  render?: (item: EventItem) => void;
+}
+
+export class LoadMoreTable extends React.PureComponent<BasicTableProps, BasicTableState> {
+  public readonly state = {
+    isPopoverOpen: false,
+  };
+
+  public render() {
+    const {
+      columns,
+      hasNextPage,
+      itemsPerRow,
+      loading,
+      loadingTitle,
+      pageOfItems,
+      title,
+      loadMore,
+      limit,
+      updateLimitPagination,
+    } = this.props;
+
+    if (loading && isEmpty(pageOfItems)) {
+      return (
+        <LoadingPanel
+          height="auto"
+          width="100%"
+          text={`Loading ${loadingTitle ? loadingTitle : title}`}
+        />
+      );
+    }
+
+    const button = (
+      <EuiButtonEmpty
+        size="s"
+        color="text"
+        iconType="arrowDown"
+        iconSide="right"
+        onClick={this.onButtonClick.bind(this)}
+      >
+        Rows: {limit}
+      </EuiButtonEmpty>
+    );
```

Comment:
This would read better yoinked out of the render and in its own file or above this render as a pure rather than having component definitions mixed in a render. The `onButtonClick bind` looks like it would be cleaned up if this was pulled out.

Reply:
I'm actually surprised you needed the `this.onButtonClick.bind(this))` looking below the `closePopover()` doesn't have a bind and the you used the correct definition to get auto binding to both functions.

Wondering if that bind is needed or not (in general)